src/libp2p_framed_stream.erl

```diff
@@ -132,10 +147,42 @@ info(Pid) ->
 %%
 %% Common
 %%
-
--spec init_module(atom(), atom(), libp2p_connection:connection(), [any()]) -> {ok, #state{}} | {error, term()}.
-init_module(Kind, Module, Connection, Args) ->
+-spec pre_init(atom(), atom(), libp2p_connection:connection(), [any()], pid() | undefined) -> {ok, #state{}}
+                                                                                              | {error, term()}.
+pre_init(Kind, Module, Connection, Args, Parent) ->
     SendPid = spawn_link(libp2p_connection:mk_async_sender(self(), Connection)),
+    case proplists:get_value(secured, Args, false) of
+        Swarm when is_pid(Swarm) ->
```

Comment:
There must be a better way to get the swarm handle here, no? maybe @madninja can advise?